### ISSUE TITLE
Import Garmin Edge 25 file

### ISSUE THREAD

**Issue by Scottfa** [GoldenCheetah/GoldenCheetah#4019]
Issue tracker is **only** for Bugs and Features, please don't open issues for questions or technical support. Before to open a new issue please read the contributing guidelines (link below).

If you have questions, please read the FAQs and User's/Developer's Guide:
* FAQs - https://github.com/GoldenCheetah/GoldenCheetah/wiki/FAQ
* User's Guide - https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Main-Page_Users-Guide
* Developer's Guide - https://github.com/GoldenCheetah/GoldenCheetah/wiki/Developers-guide

If you need help or technical support please use the forums:
* Users - https://groups.google.com/forum/#!forum/golden-cheetah-users
* Developers - https://groups.google.com/forum/#!forum/golden-cheetah-developers

**Comment by Scottfa** [comment]
Edge is in blue, Wahoo Bolt in purple. Same ride.
Wow look at the crazy graph! Also note in bottom right corner that the average speed for the Edge is 47 MPH! 
in the Summary page is shows total time is 21 minutes, even though the graph is for over an hour. Two other programs(GPXsee and Ridewith GPS) import normally.
![Edge 25 fiit file crazy](https://user-images.githubusercontent.com/89270521/130269793-0bafc8d8-5371-434e-abd0-c842755d153e.JPG)

**Comment by gille** [comment]
Can you attach the two files so I can use them to debug & verify?

**Comment by Scottfa** [comment]
Thanks!The B8K …. File is from the Edge 25. Sent from Mail for Windows 10 From: Magnus GilleSent: Friday, August 20, 2021 2:07 PMTo: GoldenCheetah/GoldenCheetahCc: Scottfa; AuthorSubject: Re: [GoldenCheetah/GoldenCheetah] Import Garmin Edge 25 file (#4019) Can you attach the two files so I can use them to debug & verify?—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by liversedge** [comment]
Looks like smart recording is enabled on the garmin?
https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Preferences_General

**Comment by Scottfa** [comment]
I have no idea. I have had several issues with my Wahoo Bolt and using a speed sensor. Also weird dropouts when GC displays the fit file, so I used an old Garmin 25 on my last ride. I can check that I guess, if you know how I can check please let me know.

**Comment by Scottfa** [comment]
I am trying to set up a chat with Garmin, but looking at my settings is not possible to set the recording method to "Smart" or anything else for that matter. Hopefully, someone at Garmin will tell me the method thecEdge 25 uses, but I suspect it is the "one second" way.

**Comment by Scottfa** [comment]
Damn, Garmin Support says the Edge 25 ONLY records in Smart mode…… Sent from Mail for Windows 10 From: Mark LiversedgeSent: Friday, August 20, 2021 2:20 PMTo: GoldenCheetah/GoldenCheetahCc: Scottfa; AuthorSubject: Re: [GoldenCheetah/GoldenCheetah] Import Garmin Edge 25 file (#4019) Looks like smart recording is enabled on the garmin?https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Preferences_General—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Scottfa** [comment]
Looks like it only shows a goofy graph when in the Compare box, even if there is no other file to compare it with. The Summary page is still really bad(i.e. 47MPH average with a max speed of 28MPH) and recording time of 21:22 when displaying the whole 1:05 ride. Sent from Mail for Windows 10 From: Mark LiversedgeSent: Friday, August 20, 2021 2:20 PMTo: GoldenCheetah/GoldenCheetahCc: Scottfa; AuthorSubject: Re: [GoldenCheetah/GoldenCheetah] Import Garmin Edge 25 file (#4019) Looks like smart recording is enabled on the garmin?https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Preferences_General—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Scottfa** [comment]
Hi:This is my second attempt at this reply, hopefully you did NOT get the first one!The Edge 25 graph only messes up in the compare pane. It does not matter if there is another file to compare, it still look screwy. The summary data is still wacky: Average speed is 45 MPH, but the Max speed is 28MHP!. Also the summary states that the recorded time is around 21 minutes, but the graph shows the correct 1:05 or so.Thanks Sent from Mail for Windows 10 From: Mark LiversedgeSent: Friday, August 20, 2021 2:20 PMTo: GoldenCheetah/GoldenCheetahCc: Scottfa; AuthorSubject: Re: [GoldenCheetah/GoldenCheetah] Import Garmin Edge 25 file (#4019) Looks like smart recording is enabled on the garmin?https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Preferences_General—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by amtriathlon** [comment]
Low end Garmin devices tend to use smart recording by default, or as the only option, you need to enable Smart Recoding in GoldenCheetah settings before import or to use Fix Gaps afterwards, see https://github.com/GoldenCheetah/GoldenCheetah/wiki/UG_Special-Topics_Activity-Processing#edit-tool-fix-gaps-in-recording and, please, use the forum for this kind of questions
